Get rid of some compilation warnings.

shlomif opened the pull request:
These appear by default on Mageia Linux v6. Not all compilation warnings
were fixed but it's a start.

bojidar-bg commented on modules/gdscript/gd_editor.cpp:
```diff
@@ -2645,6 +2645,8 @@ Error GDScriptLanguage::lookup_code(const String& p_code, const String& p_symbol
 
+		case GDParser::COMPLETION_GET_NODE: {
+		} break;
 		case GDParser::COMPLETION_NONE: {
```
You can do better here with:
```c++
case GDParser::COMPLETION_GET_NODE:
case GDParser::COMPLETION_NONE: {
} break;
```

bojidar-bg commented on main/tests/test_gdscript.cpp:
```diff
@@ -318,6 +318,7 @@ static void _parser_show_block(const GDParser::BlockNode *p_block,int p_indent)
 						else
 							_print_indent(p_indent,"return ");
 					} break;
+					case GDParser::ControlFlowNode::CF_MATCH: break;
```
That one is wrong, it should "unparse" the `match` statement, instead of just covering the issue in a break. If you think it would be too complex for the scope of this PR, you should either leave a `TODO` comment, or not change it at all.

akien-mga commented:
Hi @shlomif, nice to see a fellow Mageia dev here ;)

shlomif commented:
Thanks for the commentary @bojidar-bg ! I'll try to tend to it soon.

shlomif committed:
Fix some compilation warnings.

Redone the commit based on the input in
https://github.com/godotengine/godot/pull/7851 . Not all warnings were
fixed but it's a start.

shlomif commented:
OK , I redid the commit per @bojidar-bg 's comments.
